### PR TITLE
treewide: prepare separation of node installers

### PR DESCRIPTION
**Pull request by msanft**
To support multiple deployment "flavours" (i.e. AKS and bare-metal QEMU on TDX), we need to provide separate node installer deployments, as both deployment scenarios require different components to be installed on the hosts. This prepares this separation, by moving the Microsoft-specific runtime-class-files into the `microsoft` package set, and including the "flavour" of the deployment in the node installer image name.

### CODE REVIEW

File: .github/workflows/e2e_getdents.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
           just get-credentials
       - name: Build and prepare deployments
         run: |
-          just node-installer
+          just microsoft-node-installer
```
Comment (katexochen) [inline .github/workflows/e2e_getdents.yml:51]: please change to node-installer-microsoft, which is more intuitive for tab completion.

File: internal/kuberesource/parts.go [hunk 2]
```diff
@@ -42,12 +42,12 @@ func NodeInstaller(namespace string) *NodeInstallerConfig {
 			).
 			WithTemplate(PodTemplateSpec().
 				WithLabels(map[string]string{"app.kubernetes.io/name": name}).
-				WithAnnotations(map[string]string{"contrast.edgeless.systems/pod-role": "contrast-node-installer"}).
+				WithAnnotations(map[string]string{"contrast.edgeless.systems/pod-role": "contrast-node-installer-microsoft"}).
```
Comment (katexochen) [inline internal/kuberesource/parts.go:45]: Why do we need to change this?

**Review by katexochen — CHANGES_REQUESTED** [review]
Commit title should not end with a dot.

**Comment by msanft** [comment]
Closing in favor of #611

**Pull request closed without merging**